### PR TITLE
prov/gni: get fi_getinfo_test working again

### PR DESCRIPTION
**Pull request by hppritcha**
Get fi_getinfo_test to work with GNI provider

Fixes ofi-cray/libfabric-cray#998

@sungeunchoi 
Signed-off-by: Howard Pritchard howardp@lanl.gov

### CODE REVIEW

**Comment by sungeunchoi** [comment]
I guess this is fine, but is this basically nonsense for our provider?  What implications are there for setting it?

**Comment by jswaro** [comment]
Do we really want to allocate memory if it doesn't hit that 'else' condition? If an application reads it, without libfabric going through that code path, the contents will be undefined. 

Edit: Nevermind, I expanded the section for more context. It looks fine.

**Comment by hppritcha** [comment]
travis seems to be having download issues

**Comment by hppritcha** [comment]
It isn't exactly non-sense, but since there's no way at this point in the code to allocate a `cdm_id` (and we would not want to) the address isn't directly usable.  If we supported multiple aries per node we could in principle use it for selecting the device.  But unless you know otherwise, I don't think cray is going to support nodes with multiple aries devices.  

However, I do think I want to beef up a bit our error checking for address format being passed in.  We should only be supporting `FI_ADDR_GNI` or `FI_ADDR_UNSPEC`.  I'll add such a check to the PR.

File: prov/gni/src/gnix_ep.c [hunk 1]
```diff
@@ -1958,7 +1959,11 @@ DIRECT_FN int gnix_ep_open(struct fid_domain *domain, struct fi_info *info,
 	}
 
 	if (GNIX_EP_RDM_DGM(ep_priv->type)) {
-		if (info->src_addr != NULL) {
+		name = (struct gnix_ep_name *)info->src_addr;
+		if ((name != NULL) &&
+			(name->name_type == GNIX_EPN_TYPE_BOUND)) {
+			name = (struct gnix_ep_name *)info->src_addr;
```
Comment (sungeunchoi) [inline prov/gni/src/gnix_ep.c]: What is this line for?

**Comment by hppritcha** [comment]
@sungeunchoi fixed!

**Comment by sungeunchoi** [comment]
👍

**Pull request merged**